bytecomp/translcore.ml

```diff
@@ -521,6 +521,22 @@ and transl_exp0 e =
          }
   | Texp_unreachable ->
       raise (Error (e.exp_loc, Unreachable_reached))
+  | Texp_open (od, e) ->
+    begin match od.open_expr.mod_desc with
+    | Tmod_ident _ ->
+      transl_exp e
+    | _ ->
```

Comment:
It'd be good to handle the `open struct ... end` case specially here as well.

Reply:
Just as in `Translmod.transl_struct` I think it's actually slightly annoying to relate the internal and external idents, so I'll handle this case at the same time as the other ones (i.e. later).